gqcp/include/Operator/SecondQuantized/USQHamiltonian.hpp

```diff
@@ -188,11 +188,23 @@ class USQHamiltonian {
 
 
     /**
-     *  @param component                    the spin component
+     *  In-place rotate the matrix representation of one of the spin components of the Hamiltonian using a unitary Jacobi rotation matrix constructed from the Jacobi rotation parameters. Note that this function is only available for real (double) matrix representations
+     *
+     *  @param jacobi_rotation_parameters       the Jacobi rotation parameters (p, q, angle) that are used to specify a Jacobi rotation: we use the (cos, sin, -sin, cos) definition for the Jacobi rotation matrix
+     *  @param sigma                            the spin sigma component
+     */
+    template <typename Z = Scalar>
+    enable_if_t<std::is_same<Z, double>::value> rotate(const JacobiRotationParameters& jacobi_rotation_parameters, const Spin& sigma) {
+        this->transform(jacobi_rotation_parameters, sigma);
+    }
+
+
+    /**
+     *  @param sigma                    the spin sigma component
      * 
      *  @return the pure contributions of the requested component of the unrestricted Hamiltonian 
      */
-    const SQHamiltonian<Scalar>& spinHamiltonian(const Spin& component) const { return this->sq_hamiltonians[component]; }
+    const SQHamiltonian<Scalar>& spinHamiltonian(const Spin& sigma) const { return this->sq_hamiltonians[sigma]; }
```

Comment:
Thanks for renaming this! This shows that you're also trying to optimize the readability of the code, making it more and more readable every day!